glyphs-reader/src/font.rs

```diff
@@ -1702,31 +1658,136 @@ impl TryFrom<RawFont> for Font {
+        let raw_content =
+            preprocess_unparsed_plist(fs::read_to_string(glyphs_file).map_err(Error::IoError)?);
 
-        let mut raw_content = Plist::parse(&raw_content)
+        let raw_content = Plist::parse(&raw_content)
             .map_err(|e| Error::ParseError(glyphs_file.to_path_buf(), format!("{e:#?}")))?;
 
-        // Fix any issues with the raw plist
-        let Plist::Dictionary(ref mut root_dict) = raw_content else {
+        let raw_font = RawFont::from_plist(raw_content);
+        raw_font.try_into()
+    }
+
+    pub fn load_package(glyphs_package: &path::Path) -> Result<Font, Error> {
+        if !glyphs_package.is_dir() {
+            return Err(Error::NotAGlyphsPackage(glyphs_package.to_path_buf()));
+        }
+        debug!("Read {glyphs_package:?}");
+
+        let fontinfo_file = glyphs_package.join("fontinfo.plist");
+        let fontinfo_data = fs::read_to_string(&fontinfo_file).map_err(Error::IoError)?;
+        let mut font_plist = Plist::parse(&fontinfo_data)
+            .map_err(|e| Error::ParseError(fontinfo_file.to_path_buf(), format!("{e:#?}")))?;
+
+        let Plist::Dictionary(ref mut root_dict) = font_plist else {
             return Err(Error::ParseError(
-                glyphs_file.to_path_buf(),
+                fontinfo_file.to_path_buf(),
                 "Root must be a dict".to_string(),
             ));
         };
-        fix_glyphs_named_infinity(glyphs_file, root_dict)?;
 
-        let raw_font = RawFont::from_plist(raw_content);
+        let mut glyphs: HashMap<String, Plist> = HashMap::new();
+        let glyphs_dir = glyphs_package.join("glyphs");
+        if glyphs_dir.is_dir() {
+            for entry in fs::read_dir(glyphs_dir).map_err(Error::IoError)? {
+                let entry = entry.map_err(Error::IoError)?;
+                let name = entry.file_name();
+                if name.to_string_lossy().ends_with(".glyph") {
+                    let glyph_data = preprocess_unparsed_plist(
+                        fs::read_to_string(&entry.path()).map_err(Error::IoError)?,
+                    );
+                    let glyph_plist = Plist::parse(&glyph_data)
+                        .map_err(|e| Error::ParseError(entry.path(), format!("{e:#?}")))?;
+                    let Plist::Dictionary(ref glyph_dict) = glyph_plist else {
+                        return Err(Error::ParseError(
+                            entry.path(),
+                            "Glyph must be a dict".to_string(),
+                        ));
+                    };
+                    let glyph_name = glyph_dict
+                        .get("glyphname")
+                        .map(|s| s.to_string())
+                        .ok_or_else(|| {
+                            Error::ParseError(
+                                entry.path(),
+                                "Glyph dict must have a 'glyphname' key".to_string(),
+                            )
+                        })?;
```

Comment:
if you rebase on #450 I hope you can now just write, 
```suggestion
                    let glyph_dict = Plist::parse(&glyph_data)
                        .and_then(Plist::expect_dict)
                        .map_err(|e| Error::ParseError(entry.path(), e.to_string()))?;

                    let glyph_name = glyph_dict
                        .get("glyphname")
                        .map(|s| s.expect_string().ok())
                        .ok_or_else(|| {
                            Error::ParseError(
                                entry.path(),
                                "Glyph dict must have a 'glyphname' key".to_string(),
                            )
                        })?;
```

which... is still pretty gross!